session/session.go

```diff
@@ -2530,6 +2531,26 @@ func (s *session) checkPlacementPolicyBeforeCommit() error {
 					fmt.Sprintf("table or partition %v's leader location %v is out of txn_scope %v", physicalTableID, dcLocation, txnScope))
 				break
 			}
+			// FIXME: currently we assumes the physicalTableID is the partition ID. In future, we should consider the situation
+			// if the physicalTableID belongs to a Table.
+			if is.IsPartitionID(physicalTableID) {
```

Comment:
I don't think this check is needed, because it adds one more search, which is similar to `FindTableByPartitionID`.
When `tbl` returned by `FindTableByPartitionID` is `nil`, you can assume that the physical id is table id.